online-survey-app/src/app/tangy-forms/tangy-forms-player/tangy-forms-player.component.ts

```diff
@@ -52,6 +57,24 @@ export class TangyFormsPlayerComponent implements OnInit {
   async ngOnInit(): Promise<any> {
     this.window = window;
 
+    // we are using the query parameters to get the actor and auth information
+    this.route.queryParamMap.subscribe((query) => {
+      const actorRaw = query.get('actor');
+      const authRaw = query.get('auth');
+
+      this.endpoint = query.get('endpoint');
+      this.registration = query.get('registration');
+
+      // here we parse the auth and actor query parameter
+      try {
+        this.auth = JSON.parse(decodeURIComponent(authRaw))[0];
```

Comment:
```decodeURIComponent``` is not necessary / should not be used as per my understanding of the docs. 

https://angular.dev/api/router/ParamMap

States that this provides URLSearchParams. URLSearchParams takes care of decoding itself.